src/panels/config/entities/entity-registry-settings.ts

```diff
@@ -1069,11 +1118,28 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
       params.hidden_by = this._hiddenBy;
     }
     if (
-      (domain === "number" || domain === "sensor") &&
-      stateObj?.attributes?.unit_of_measurement !== this._unit_of_measurement
+      ((domain === "number" || domain === "sensor") &&
+        stateObj?.attributes?.unit_of_measurement !==
+          this._unit_of_measurement) ||
+      (domain === "sensor" &&
+        this.entry.options?.[domain]?.precision !== this._precision)
     ) {
       params.options_domain = domain;
-      params.options = { unit_of_measurement: this._unit_of_measurement };
+      params.options = this.entry.options?.[domain] || {};
+      if (
+        stateObj?.attributes?.unit_of_measurement !== this._unit_of_measurement
+      ) {
+        params.options = {
+          ...params.options,
+          unit_of_measurement: this._unit_of_measurement,
+        };
+      }
+      if (
+        domain === "sensor" &&
+        this.entry.options?.[domain]?.precision !== this._precision
+      ) {
+        params.options = { ...params.options, precision: this._precision };
+      }
```

Comment:
Can we keep thing separated ? One block of code for number/sensor + unit and one block for sensor + precision.